server/aesdsocket.c

```diff
@@ -0,0 +1,376 @@
+        	if (accepted_fd == -1) 
+        	{
+            		syslog(LOG_WARNING, "WARNING: Failed to accept, retrying ...");
+            	continue; // Continue accepting connections
+        	}
+        	syslog(LOG_INFO, "connection accepted: %d",accepted_fd);
+        	
+        	// Log accepted connection
+        	// converts binary ip address to string format
+        	char client_ip[INET_ADDRSTRLEN];
+        	if (inet_ntop(AF_INET, &(client_addr.sin_addr), client_ip, INET_ADDRSTRLEN) == NULL)
+        	{
+        		syslog(LOG_ERR, "ERROR: Failed to get ip");	
+        	}
+        	syslog(LOG_INFO, "Accepted connection from %s", client_ip);
+        	
+        	// Open file for aesdsocketdata
+    		file_fd = open(FILENAME, O_CREAT | O_RDWR | O_APPEND, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+
+    		if (file_fd == -1)
+    		{
+        		syslog(LOG_ERR, "ERROR: Failed to create/open file");
+        		close_n_exit(EXIT_FAILURE);
+    		}
+    		syslog(LOG_INFO, "File open successfully :%d",file_fd);
+        	
+        	do
+        	{
+        		memset(buffer, 0, BUFF_SIZE);
+            		// recieve data from client
+            		recv_bytes = recv(accepted_fd, buffer, BUFF_SIZE, 0);
+            		if (recv_bytes == -1)
+            		{
+                		syslog(LOG_ERR, "ERROR: Failed to recieve byte from client");
+        			close_n_exit(EXIT_FAILURE);
+            		}
+            		syslog(LOG_INFO, "data recieved");
+            		
+            		// write the string received to the file 
+            		written_bytes = write(file_fd, buffer, recv_bytes);
+            		if (written_bytes != recv_bytes)
+            		{
+                		syslog(LOG_ERR, "ERROR: Failed to write bytes to file");
+        			close_n_exit(EXIT_FAILURE);
+            		}
+            		syslog(LOG_INFO, "data write to file");
+            		
+            		/* check for new line */
+            		if (NULL != (memchr(buffer, '\n', recv_bytes)))
+            		{
+                		packet_complete = true;
+            		}
+        		
+        	}while(!packet_complete);
+        	
+        	packet_complete = false;
+
+        	// Set file pos to begining of file
+        	off_t offset = lseek(file_fd, 0, SEEK_SET);
+        	if (offset == -1)
+        	{
+        		syslog(LOG_ERR, "ERROR: Failed to SET file offset");
+        		close_n_exit(EXIT_FAILURE);
+        	}
+        	syslog(LOG_INFO, "lseek set");
+        	
+        	/* read file contents till EOF */
+        	int read_bytes = 0;
+        	int send_bytes = 0;
+        	do
+        	{
+        		memset(buffer, 0, BUFF_SIZE);
+            		read_bytes = read(file_fd, buffer, BUFF_SIZE);
+            		if (read_bytes == -1)
+            		{
+                		syslog(LOG_ERR, "ERROR: Failed to read from file");
+        			close_n_exit(EXIT_FAILURE);
+            		}
+            		syslog(LOG_INFO, "read succesful is: %d", read_bytes);
+            		syslog(LOG_INFO, "read succesful is: %s", buffer);
```

Comment:
@JayashCUB I don't think `buffer` is guaranteed null terminated here if you read `BUFF_SIZE` in line 219.  I think that may be what https://github.com/cu-ecen-aeld/assignments-3-and-later-JayashCUB/pull/3/files#diff-85261e472d268426712db558b85101dca2eb5ab21de43bcc50c1450ee339e7ecR140 is telling you